src/systems/bit-physics.ts

```diff
@@ -0,0 +1,108 @@
+import { defineQuery, enterQuery, entityExists, exitQuery, hasComponent, Not } from "bitecs";
+import {
+  Object3DTag,
+  Rigidbody,
+  PhysicsShape,
+  AEntity,
+  AmmoShape,
+  BoxCollider,
+  Trimesh,
+  HeightField
+} from "../bit-components";
+import { Fit, getShapeFromPhysicsShape, PhysicsShapes } from "../inflators/physics-shape";
+import { findAncestorWithComponent, hasAnyComponent } from "../utils/bit-utils";
+import { Mesh, Vector3, Object3D } from "three";
+import { getBodyFromRigidBody } from "../inflators/rigid-body";
+import { HubsWorld } from "../app";
+import { PhysicsSystem } from "./physics-system";
+
+const rigidbodyQuery = defineQuery([Rigidbody, Object3DTag, Not(AEntity)]);
+const rigidbodyEnteredQuery = enterQuery(rigidbodyQuery);
+const rigidbodyExitedQuery = exitQuery(rigidbodyQuery);
+const shapeQuery = defineQuery([PhysicsShape]);
+const shapeEnterQuery = enterQuery(shapeQuery);
+const shapeExitQuery = exitQuery(shapeQuery);
+
+const tmpV = new Vector3();
+
+function updateOffsets(world: HubsWorld, eid: number, obj: Object3D) {
```

Comment:
The reason for this is that when adding objects without a Rigid body, we default to the closest parent rigid body and if we don't set offsets the shape will use the parent transform so the shapes will be incorrectly positioned. This is usually the same if you add an `ammo-shape` to a scene.

For low density scenes the old loader adds a `Fit.ALL` `Mesh` collider in `traverseMeshesAndAddShapes` for all the environment meshes but it's not respecting the users design in case there are `ammo-shapes`. For high density scenes the old loader only sets a floor `Box` collider and all the ammo shapes are wrongly placed at the scene origin.

Reply:
We should check if this is correctly handled in the old code. If not (probably true) I doubt anyone is actually using the components. We might just keep the old broken behavior and can think through how to simplify.

Reply:
Yes, this is also broken in the old code and I agree that probably people don't use it. I thought it would be worth fixing but if it's not that simplifies the code and we can remove all those tag components.

Reply:
Yeah if its not a regression I think we should fix it by thinking a bit more about what we want to be configurable in Blender.